hpxc/util/init_hpx.hpp

```diff
@@ -14,5 +14,13 @@ int hpxc_user_main(int argc, char** argv);
 #endif
 
 #if !defined(HPXC_EXPORTS)
-#define main(...) hpxc_user_main(__VA_ARGS__)
+
+// Dispatch differently depending on the number of arguments, such that hpxc_user_main always gets defined with two arguments
+#define HPXC_MAIN0() hpxc_user_main(int _HPXC_UNUSED_ARGC, char** _HPXC_UNUSED_ARGV)
+#define HPXC_MAIN1(...) hpxc_user_main(int _HPXC_UNUSED_ARGC, char** _HPXC_UNUSED_ARGV)
+#define HPXC_MAIN2(...) hpxc_user_main(__VA_ARGS__)
+
+#define HPXC_GET_MAIN_SIGNATURE(_1,_2, NAME,...) NAME
+#define main(...) HPXC_GET_MAIN_SIGNATURE(__VA_ARGS__ __VA_OPT__(,) HPXC_MAIN2, HPXC_MAIN1, HPXC_MAIN0)(__VA_ARGS__)
```

Comment:
You could have used `HPX_PP_NARGS` (https://github.com/STEllAR-GROUP/hpx/blob/master/libs/core/preprocessor/include/hpx/preprocessor/nargs.hpp)